internal/webhook/handlers/tcp_defaults.go

```diff
@@ -60,6 +74,10 @@ func (t TenantControlPlaneDefaults) OnUpdate(object runtime.Object, oldObject ru
 			return nil, fmt.Errorf("DataStore is a required field")
 		}
 
+		if oldTCP.Spec.ControlPlane.Deployment.ServiceAccountName != newTCP.Spec.ControlPlane.Deployment.ServiceAccountName {
+			return nil, fmt.Errorf("changing serviceAccountName is not supported")
+		}
+
```

Comment:
What's the rationale here? A Pod can easily switch from a ServiceAccount to another.

Reply:
Kamaji doesn't manage Pod directly, that's done by Deployment which allows to replace ServiceAccounts.